nms/app/packages/magmalte/app/views/subscriber/SubscriberStateTable.js

```diff
@@ -0,0 +1,263 @@
+        {
+          title: 'Active Duration',
+          field: 'activeDuration',
+        },
+        {
+          title: 'Active Policy IDs',
+          field: 'activePolicies',
+          render: currRow =>
+            currRow.activePolicies.length ? (
+              <List>
+                {currRow.activePolicies.map(policy => (
+                  <ListItem key={policy.id}>
+                    <Link>{policy.id} </Link>
+                  </ListItem>
+                ))}
+              </List>
+            ) : (
+              <Text>{'-'}</Text>
+            ),
+        },
+      ]}
+      options={{
+        actionsColumnIndex: -1,
+        pageSizeOptions: [5],
+        toolbar: false,
+        paging: false,
+        rowStyle: {background: '#f7f7f7'},
+        headerStyle: {
+          background: '#f7f7f7',
+          color: colors.primary.comet,
+        },
+      }}
+    />
+  );
+}
+
+export default function SubscriberStateTable() {
+  const [currRow, setCurrRow] = useState<SubscriberRowType>({});
+  const classes = useStyles();
+  const ctx = useContext(SubscriberContext);
+  const subscriberMap: {[string]: subscriber} = ctx.state;
+  const sessionState: {[string]: subscriber_state} = ctx.sessionState;
+  const subscriberMetrics = ctx.metrics;
+  const [jsonDialog, setJsonDialog] = useState(false);
+  const subscribersIds = Object.keys(sessionState);
+  const networkCtx = useContext(NetworkContext);
+
+  const tableData: Array<SubscriberRowType> = subscribersIds.map(
+    (imsi: string) => {
+      const subscriberInfo = subscriberMap[imsi] || {};
+      const metrics = subscriberMetrics?.[`${imsi}`];
+      const subscriber = sessionState[imsi]?.subscriber_state || {};
+      const ipAddresses = [];
+      const activeApns = [];
+      let activeSessions = 0;
+      Object.keys(subscriber || {}).forEach(apn => {
+        subscriber[apn].forEach(session => {
+          if (session.lifecycle_state === 'SESSION_ACTIVE') {
+            ipAddresses.push(session?.ipv4);
+            activeSessions++;
+          }
+        });
+        activeApns.push(apn);
+      });
+      return {
+        name: subscriberInfo.name ?? imsi,
+        imsi: imsi,
+        service: subscriberInfo.lte?.state || '',
+        currentUsage: metrics?.currentUsage ?? '0',
+        activeApns: activeApns.length > 0 ? activeApns.join() : '-',
+        activeSessions: activeSessions,
+        ipAddress: ipAddresses.length > 0 ? ipAddresses.join() : '-',
+        dailyAvg: metrics?.dailyAvg ?? '0',
+        lastReportedTime:
+          subscriberInfo.monitoring?.icmp?.last_reported_time === 0
+            ? new Date(subscriberInfo.monitoring?.icmp?.last_reported_time)
+            : '-',
+      };
+    },
+  );
+
+  const tableColumns = [
+    {
+      title: 'Name',
+      field: 'name',
+    },
+    {
+      title: 'IMSI',
+      field: 'imsi',
+      render: currRow => {
+        const subscriberConfig = subscriberMap[currRow.imsi];
```

Comment:
Hey @HannaFar , given that we don't have subscriberMap populated with all IMSIs, won't subscriberConfig be empty here ? We can perhaps avoid displaying it as a link when a user attempts views the subscriber state table.